chapters/en/slides/module2/module2_03.Rmd

```diff
@@ -0,0 +1,284 @@
+to be readable in a vertical bar chart.
+
+Since this is the case for our plot
+we will continue to use horizontal bar charts
+in this slide deck.
+
+---
+
+## Visualizing counts of values with bar charts
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y='region')
+```
+
+Notes:
+To count values,
+we can use the aggregation function `count()` in Altair.
+We don't need to specify a column name for the y-axis,
+since we are just counting values in each categorical group on the y-axis.
+
+---
+
+## Ordering bars makes them easier to interpret
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y=alt.Y('region'))
+```
+
+Notes:
+Unless the categorical axis has a natural order to it,
+it is best to sort bar charts from high to low values.
+This makes it easier to see any trends in the data,
+and to compare bars of similar height more carefully.
+
+To sort the bars,
+we will use the helper functions `alt.X` and `alt.Y`,
+instead of `x=` and `y=`.
+These have the role of customizing things like order, groups, and scale for the axis.
+When using just `y='column'`,
+we're still calling `alt.Y()` under the hood,
+we just save ourselves some typing.
+
+In this slide,
+we use `alt.Y`
+on its own without any additional argument
+to recreate the same plot as above
+just to demonstrate how it works.
+
+---
+
+## Ordering bars makes them easier to interpret
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y=alt.Y('region', sort='x'))
+```
+
+Notes:
+Sorting can now be done via the `sort` parameter in `alt.Y`.
+In this example,
+we're typing `sort='x'` to specify that we want to sort
+according to the values on the y-axis.
+
+When sorting by value,
+it is often more visually appealing with the longest bar
+the closet to the axis line,
+as in this slide.
+
+---
+
+## The longest bar should be the closest to the axis line
+
+```{python}
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y=alt.Y('region', sort='-x'))
+```
+
+If we for some reason would want the reverse order,
+we could use the minus sign before the axis reference.
+
+---
+
+## Sorting in a custom order
+
+```{python}
+my_order = ['Africa', 'Europe', 'Oceania', 'Asia', 'Americas']
+alt.Chart(gm2018).mark_bar().encode(
+    x='count()',
+    y=alt.Y('region', sort=my_order))
+```
+
+Notes:
+Sometimes there is a natural order to our data
+that we want to use for the bars,
+for example days of the week.
+
+For situations like this,
+we can pass a list or array
+to the `sort` parameter.
+
+We can either create this list manually as in thie slide,
+or use the pandas sort function if we wanted
+for example reverse alphabetical order.
+
+To learn more about good considerations
+when plotting counts of categorical observations,
+you can refer to [chapter 6 of Fundamental of Data Visualizations](https://clauswilke.com/dataviz/visualizing-amounts.html).
+
+---
+
+## Using bars to show counts of quantitative values
```

Comment:
Could we introduce histograms first with discrete data, and then continuous? Maybe an example where there are bins for just each value, like on a dice? and then we could talk about binning in this case, and then move into why we need binning for quantitative/non-discrete data?

Reply:
I also feel like this section is missing a real life example as to why you would plot a histogram - we need a question to answer with this example!

Reply:
I am considering moving this entire histogram section into module 3 where we talk about distibutions, but I am not sure if it wuold mess up the assignment, @hfboyce ?

And I like the example with the dice or similar, that would be a nice transition

Reply:
@hfboyce Just fyi, I don't think I will move histograms, there will just be some repetition in the next module.

Reply:
Added question and mentioned dice histograms.